src/api/ans.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Account, AccountAddress } from "../core";
+import { getOwnerAddress, registerDomain } from "../internal/ans";
+import { InputSingleSignerTransaction, InputGenerateTransactionOptions } from "../transactions/types";
+import { HexInput } from "../types";
+import { AptosConfig } from "./aptosConfig";
+
+/**
+ * A class to handle all `ANS` operations
+ */
+export class ANS {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Retrieve the owner address of a domain name or subdomain name.
+   *
+   * ```ts
+   * getOwnerAddress({domainName: "aptos", subdomainName: "test"})
+   * // Will return the owner address of "test.aptos.apt" or undefined
+   * ```
+   *
+   * @param args.domainName - A string of the domain name to retrieve
+   * @param args.subdomainName - A string of the subdomain name to retrieve
+   *
+   * @returns MoveAddressType if the name is owned, undefined otherwise
+   */
+  async getOwnerAddress(args: { domainName: string; subdomainName?: string }): Promise<AccountAddress | undefined> {
+    return getOwnerAddress({ aptosConfig: this.config, ...args });
+  }
```

Comment:
I stared at this a bit.

Can we make an easy mode, where we take in say... `test.greg.apt` and parse it out into it's parts?

Otherwise, the dapp or dev will have to write a wrapper to do this every time they take an ANS name from a user.  I doubt that they'll already be split into pieces.